prompt-service/src/unstract/prompt_service/helper.py

```diff
@@ -80,3 +81,55 @@ def plugin_loader(app: Flask) -> dict[str, dict[str, Any]]:
         app.logger.info("No plugins found.")
 
     return plugins
+
+
+def query_usage_details(db, run_id, token):
+    org_id = AuthenticationMiddleware.get_account_from_bearer_token(token)
+    metadata = {"run_id": run_id}
+    query = f"""
+        SELECT
+            usage_type,
+            llm_usage_reason,
+            model_name,
+            SUM(prompt_tokens) AS input_tokens,
+            SUM(completion_tokens) AS output_tokens,
+            SUM(total_tokens) AS total_tokens,
+            SUM(embedding_tokens) AS embedding_tokens
+        FROM "{org_id}"."token_usage"
+        WHERE run_id = %s
+        GROUP BY usage_type, llm_usage_reason, model_name;
+    """
+    logger = current_app.logger
+    try:
+        with db.atomic():
+            logger.info("Querying usage metadata")
+            cursor = db.execute_sql(query, (run_id,))
+            results = cursor.fetchall()
+            # Process results as needed
+            for row in results:
+                # Each column is accessed as row[0], row[1], and so forth.
+                usage_type = row[0]
+                llm_usage_reason = row[1]
+                model_name = row[2]
+                if llm_usage_reason:
```

Comment:
too many nested.. Can you refactor it?

Reply:
@muhammad-ali-e I don't see any room for refactoring here. We can move some pieces as separate function. It won't be needed for this case because it is already a small function .

Reply:
can we do the block of code inside for loop in to another function?

Reply:
Done